envision/server.py

```diff
@@ -182,27 +184,33 @@ def stop(self):
     def run_forever(self):
         async def run_loop():
             # If no frame, wait till one is present
-            frame = self._frames.start_frame
-            while frame is None:
+            frame_ptr = self._frames.start_frame
+            while frame_ptr is None:
                 await asyncio.sleep(0.1)
-                frame = self._frames.start_frame
+                frame_ptr = self._frames.start_frame
 
+            frames_to_send = []
+            last_sent_time = 0
             while True:
                 # Handle seek
                 if self._seek is not None and self._frames.start_time is not None:
-                    frame = self._frames(self._frames.start_time + self._seek)
+                    frame_ptr = self._frames(self._frames.start_time + self._seek)
```

Comment:
True, though seek is pretty glitchy at the moment regardless. Will need to investigate why.

Reply:
Oh, it's probably because the `next_` pointer needs to be updated too.

Reply:
the `next_` pointer points to the next frame in the next timestamp. Why would that need to be changed and what would it change to?

Reply:
Oops, yeah I was wrong about that being needed.  Sorry.

Reply:
Created issue #807 for this. 

From my investigation, when jumping to a new timestamp, the server is sending a long list of frames as expected, but the timestamp between the batches is not in sync. On Envision side, the loop in simulation.js is not running after seek. 
Possible places to look at are:
- server.py when it is sending frames to client
- simulation.js loop on line 168
- `seek` and `obtainstream` in client.js. 1 possibility is the for loop to loop over received frames on line 93 in client.js will cause the front end to stuck (since javascript is single-threaded) when server is sending lots of frames